### PR TITLE
Security config dev branch

### PR DESCRIPTION
**Pull request by SebastianBoe**
see commit messages

### CODE REVIEW

**Review by vili-nordic — APPROVED** [review]

**Comment by SebastianBoe** [comment]
@vidar: Added some more unrelated commits FYI, so this is now a dev branch (but I still want review).

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c
Comment (vili-nordic) [inline platform/ext/target/nordic_nrf/common/core/target_cfg.c:1305]: Rename variable port -> pin ? Maybe?

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c
Comment (vili-nordic) [inline platform/ext/target/nordic_nrf/common/core/target_cfg.c:1311]: Is hard coding this to 2 right, or should it be ARRAY_SIZE(secure_pins) like in the previous case?

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c
Comment (SebastianBoe) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:1305]: hmm, If you have misread then I am sure I can improve somehow.

But here, port is 0 or 1 and represents P0 or P1.

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c
Comment (SebastianBoe) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:1311]: I can do ARRAY_SIZE :)

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c
Comment (vili-nordic) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:1305]: hmm. Ok, I see that now

**Comment by SebastianBoe** [comment]
@vili-nordic : Pushed 4 more commits ontill top.

still waiting for a review from someone else...

@mswarowsky , @Vge0rge

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c
Comment (mswarowsky) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:1275]: It confuses a bit at first look that the array is called xx_pins and then the iteration variable is called port

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c
Comment (mswarowsky) [inline platform/ext/target/nordic_nrf/common/core/target_cfg.c:1095]: Didn't you mean to  say `That these addresses are just  missing in MDK therefore using the addressed hard coded to configure them`

**Review by mswarowsky — APPROVED** [review]
Maybe add 
fixup! [nrf noup] platform: nordic_nrf: Add support for 54l
to all the noup's, else just a small nit, but not a blocker

**Pull request merged**
